### DISCUSSION TITLE
How do I find where a cached venv was created?

### DISCUSSION THREAD

**Discussion by MajorDallas** [python-poetry/poetry#7789] [Q&A]
I seem to have more venvs in `~/.cache/pypoetry/virtualenvs/` than I expected. The complication is that I _do_ expect some of these duplicates but not all of them.

`poetry env list` doesn't reliably list all the venvs associated to the project, presumably because some of them were instantiated in a different path. Eg., I have `submodule_project` as its own git repo directory, and I also have it in `main_project/submodule_project` where it's pulled in by git as a submodule. I've run poetry commands in both paths, so I have separate venvs for both. That's expected and not a problem.

Let's assume I have a third venv for `submodule_project`. It uses the same python version as the other two; by all appearances, it's exactly the same as the other two. I just don't know where it came from--did I have a third copy of the repo somewhere (that maybe got deleted?), did I do something weird with a poetry command that resulted in a duplicate..? The `env list` command doesn't list this one in either copy of the project. It seems to be orphaned, so I should just `rm -rf` it to reclaim the disk space.

Before I do that, how can I make _sure_ it's really orphaned? Is there somewhere I can lookup the path where the venv was created from or some command to point me to the project directory? `grep` and google have so far failed me.

**Comment by dimbleby** [comment]
you can err on the side of deleting virtual environments, safe in the knowledge that you'll get a new one if you need it.
